server/service/service.go

```diff
@@ -157,6 +162,20 @@ type EntityManager interface {
 type Service struct {
 	bpb.UnimplementedBootstrapServer
 	em EntityManager
+	// Session cache for nonces or other state for non-IDevID flows
+	sessions sync.Map // map[string]*streamSession
```

Comment:
I don't think you need to keep the session state in the Service level. It only needs to exist for the lifetime of the stream. In other words, each BootstrapStream() call will handle exactly one gRPC stream and should be deleted when the stream ends.

Reply:
good to know. makes this simpler.